pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -282,6 +283,23 @@ func CanUseLease(client discovery.DiscoveryInterface) (bool, error) {
 	return false, nil
 }
 
+// CanUseLeases returns if leases can be used for leader election. If the resource is defined in the config
+// It uses it. Otherwise it uses the discovery client for leader election.
+func CanUseLeases(client discovery.DiscoveryInterface) (bool, error) {
+	config := config.Datadog.GetString("leader_election_default_resource")
```

Comment:
maybe the variable name can be more self explained
for example 
```suggestion
resourceType := config.Datadog.GetString("leader_election_default_resource")
```